cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/ByteBufferUtils.java

```diff
@@ -222,4 +215,55 @@ public static void skipFully(InputStream is, long length) throws IOException
             throw new EOFException("EOF after " + skipped + " bytes out of " + length);
         }
     }
+
+    // Extract component position from buffer; return null if there are not enough components
+    public static ByteBuffer extractComponent(ByteBuffer buffer, int position)
+    {
+        buffer = buffer.duplicate();
+        readStatic(buffer);
+        int index = 0;
+        while (buffer.remaining() > 0)
+        {
+            ByteBuffer c = readBytesWithShortLength(buffer);
```

Comment:
Nit: I would avoid single-symbol variable names and use something like `character` instead of `c`.